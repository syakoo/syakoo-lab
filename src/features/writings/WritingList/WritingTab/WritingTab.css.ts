import { recipe } from "@vanilla-extract/recipes";
import { tokens } from "@/design-system/tokens";

export const writingTabStyle = {
  item: recipe({
    base: {
      padding: tokens.spaces[100],
      textDecoration: "none",
    },

    variants: {
      active: {
        true: {
          borderBottom: "4px solid currentColor",
          color: tokens.colors.text.primary,
        },
        false: {
          color: tokens.colors.text.secondary,
        },
      },
    },
  }),
};
