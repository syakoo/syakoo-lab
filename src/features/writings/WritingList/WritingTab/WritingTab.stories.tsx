import type { Meta, StoryObj } from "@storybook/react";

import { WritingTab } from ".";

const meta = {
  component: WritingTab,
} satisfies Meta<typeof WritingTab>;

export default meta;

type Story = StoryObj<typeof meta>;
export const Sample: Story = {
  args: {
    selectedType: "all",
  },
};
