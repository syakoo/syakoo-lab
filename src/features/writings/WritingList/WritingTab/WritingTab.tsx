import { Row } from "@/design-system/layout";
import { Link } from "@/design-system/ui";
import {
  writingListTypeConfig,
  type WritingListType,
} from "@/features/writings/WritingList/_shared/writingListType";

import { writingTabStyle } from "./WritingTab.css";

type WritingTabProps = {
  selectedType: WritingListType;
};

export const WritingTab: React.FC<WritingTabProps> = ({ selectedType }) => {
  return (
    <Row>
      {writingListTypeConfig.map(({ label, type, url }) => {
        const isActive = selectedType === type;

        return (
          <Link key={type} href={url} noHovered={isActive}>
            <div
              aria-current={isActive}
              className={writingTabStyle.item({
                active: isActive,
              })}
            >
              {label}
            </div>
          </Link>
        );
      })}
    </Row>
  );
};
