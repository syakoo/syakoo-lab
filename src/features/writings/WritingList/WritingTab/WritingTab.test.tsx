import { testStories } from "@/test/testStories";

import * as stories from "./WritingTab.stories";

describe("WritingTab", () => {
  testStories(stories);
});
