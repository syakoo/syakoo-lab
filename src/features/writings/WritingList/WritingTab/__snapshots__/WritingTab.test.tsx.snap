// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WritingTab スナップショット Sample 1`] = `
<body>
  <div>
    <div
      class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_stretch__15ivi84u Flex_flexStyle_variants_align_stretch__15ivi849 Flex_flexStyle_direction_row__15ivi84j Flex_flexStyle_gap_100__15ivi84y Flex_flexStyle_variants_gap_100__15ivi84d Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
    >
      <a
        class="Link_linkStyle__m4v9430 Link_linkStyle_noHovered_true__m4v9433"
        href="/writings"
      >
        <div
          aria-current="true"
          class="WritingTab_writingTabStyle_item__1s7we910 WritingTab_writingTabStyle_item_active_true__1s7we911"
        >
          All
        </div>
      </a>
      <a
        class="Link_linkStyle__m4v9430"
        href="/writings?type=article"
      >
        <div
          aria-current="false"
          class="WritingTab_writingTabStyle_item__1s7we910 WritingTab_writingTabStyle_item_active_false__1s7we912"
        >
          Article
        </div>
      </a>
      <a
        class="Link_linkStyle__m4v9430"
        href="/writings?type=note"
      >
        <div
          aria-current="false"
          class="WritingTab_writingTabStyle_item__1s7we910 WritingTab_writingTabStyle_item_active_false__1s7we912"
        >
          Note
        </div>
      </a>
      <a
        class="Link_linkStyle__m4v9430"
        href="/writings?type=diary"
      >
        <div
          aria-current="false"
          class="WritingTab_writingTabStyle_item__1s7we910 WritingTab_writingTabStyle_item_active_false__1s7we912"
        >
          Diary
        </div>
      </a>
    </div>
  </div>
</body>
`;
